math.go

```diff
@@ -14,12 +17,12 @@ func dct(width int, height int, values []uint8) (result []float32) {
 				for j := 0; j < width; j++ {
 
 					sum += float64(values[i*width+j]) *
-						math.Cos(((math.Pi*float64(u))/(2*float64(height)))*(2*float64(i)+1)) *
-						math.Cos(((math.Pi*float64(v))/(2*float64(width)))*(2*float64(j)+1))
+						math.Cos(((math.Pi*float64(u))/doubleHeight)*(2*float64(i)+1)) *
+						math.Cos(((math.Pi*float64(v))/doubleWidth)*(2*float64(j)+1))
 				}
 			}
 
-			result[u*width+v] = float32(sum)
+			result[u*width+v] = int16(sum)
```

Comment:
Oh sorry, I think I tricked you. If you look closely, you’ll see the example code is only using the R component. :P

Re gamma: I think it will help with those cases where there are near-white images being compared, since it might help to separate out the detail in that range. Needs testing. I don’t think it’s as valuable for doing any sort of range compression for this kind of application, but I haven’t thought too deeply about it.